libs/core/src/CutFlowManager.cpp

```diff
@@ -68,17 +66,24 @@ void CutFlowManager::UpdateCutFlow(string cutName) {
 }
 
 void CutFlowManager::SaveCutFlow() {
+  if (eventWriter == NULL) {
+    error() << "Error: No existing eventWriter for CutFlowManager - cannot save CutFlow\n";
+  }
+  if (!eventReader->inputFile->Get("CutFlow")) {
+    info() << "Input file doesn't contain CutFlow directory yet... will create a new one in the output file.\n";
+    eventWriter->outFile->mkdir("CutFlow");
+  } 
   eventWriter->outFile->cd("CutFlow");
 
   for (auto &[cutName, sumOfWeights] : weightsAfterCuts) {
```

Comment:
I don't see `hasInitial` used anywhere. Either it's not needed anymore (and then I'd remove it completely), or there are some cases that are no longer handled. Have you tested it in all possible combinations, i.e. when the input file has/doesn't have the CutFlow directory, the directory is empty or contains some histograms already, etc.?